js/ssai_pulse.js

```diff
@@ -237,19 +273,20 @@ OO.Ads.manager(function(_, $)
       {
         adMode = true;
         this.currentAd = ad;
-        if (this.currentAd.ad)
-        {
-          this.currentAd.ad.id3AdId = this.currentId3Object.adId;
+        if (ad.ad && ad.ad.data) {
+          this.adIdDictionary[ad.ad.data.id].curAdId = ad.id;
           _handleTrackingUrls(this.currentAd, ["impression", "start"]);
-          amc.notifyLinearAdStarted(this.currentAd.id,
-            {
-              name: this.currentAd.ad.name,
-              hasClickUrl: true,
-              duration: this.currentAd.duration,
-              ssai: this.currentAd.ad.ssai,
-              isLive: this.currentAd.ad.isLive
-            }
-          );
+          if (ad.ad.name && ad.duration && ad.ad.ssai && ad.ad.isLive) {
```

Comment:
@cacel when I asked to validate some of these fields, I did not mean to just check for existence or if they are true. In fact, this line might break the code. What I mean by data validation is to ensure that ad.duration is a number, for example. Not all fields may need to be validated this way, only some of them. I would say duration is the main one to focus on